benchmark/tests/ManyEntitiesEqual.hs

```diff
@@ -34,42 +34,74 @@ type Top n =
     => Signal System (Unsigned n)
     -> Signal System (Unsigned n)
 
+#if __GLASGOW_HASKELL__ >= 904
```

Comment:
Why not make an intermediate macro.

```C
#if __GLASGOW_HASKELL__ >= 904
#define CLASH_NOINLINE OPAQUE
#else
#define CLASH_NOINLINE NOINLINE
#endif
```

Reply:
That does make it even more work, I'd have to add `#include blaat.h` everywhere and I can't do that with a regex. But.. it would be quite a bit neater so I'll try.

Reply:
Minor bikeshedding: I think calling it `CLASH_NOINLINE` will be seen by some users and confuse them, i.e. _What is the different between `NOINLINE` and `CLASH_NOINLINE`?_ I think `CLASH_PRIMITIVE` would be a better name --- primitives are the only things which should be so aggressively left alone AFAICT

Reply:
Primitives and things with synthesize annotations.

Reply:
Malicious mode: 

```
  if impl(ghc >= 9.4)
    CPP-Options: -DNOINLINE=OPAQUE
  else
    CPP-Options: -DNOINLINE=NOINLINE
```

Reply:
I _love_ it

Reply:
I've picked a slightly different name than suggested here: `CLASH_OPAQUE`. I've also added a comment referring people to this PR to prevent confusion. The other suggestions:

* `CLASH_NOINLINE`: Uses "old" terminology, I'd prefer to use the new one.
* `CLASH_PRIMITIVE`: It's not only primitives, also top entities. Basically anything that has some annotation attached to it Clash should later pick up on.
* `NOINLINE`: Love it but no :D 

I'll unresolve the conversation so it is easier to spot for reviewers.

Reply:
I'm happy with that! `CLASH_INLINE` was just something I thought at in the second I posted the comment.